var/spack/repos/builtin/packages/squashfs/package.py

```diff
@@ -85,9 +91,13 @@ class Squashfs(MakefilePackage):
 
     depends_on("zlib-api", when="+gzip")
     depends_on("lz4", when="+lz4")
+    depends_on("lz4 libs=shared,static", when="+lz4 +static")
```

Comment:
I think setting `libs=static` should be enough.

Reply:
Also you need 

```python
depends_on("zlib~shared", when="^zlib")
```

and `zlib-ng` (the other zlib-api provider) is missing a ~/+shared toggle, so that should be added.